File: client/src/api/comment.ts
import { API } from "./index"

export interface CommentResponse {
    id: number
    user_id: number
    post_id: number
    username: string
    profile_picture?: string | null
    content: string
    created_at: string
}

export const getComments = async (postId: number): Promise<CommentResponse[]> => {
    const response = await API.get(`/comments/${postId}`)
    return response.data
}

export const createComment = async (content: string, postId: number): Promise<CommentResponse> => {
    const response = await API.post("/comments/", {
        post_id: postId,
        content
    })
    return response.data
}

File: server/src/schemas/comment.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime

class CommentCreate(BaseModel):
    content: str
    post_id: int


        
class CommentResponse(BaseModel):
    id: int
    user_id: int
    post_id: int

    username: str
    profile_picture: str | None = None

    content: str
    created_at: datetime    

    model_config = ConfigDict(from_attributes=True)